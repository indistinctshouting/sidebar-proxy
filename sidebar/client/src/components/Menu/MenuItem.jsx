import React from 'react';
import { Media } from 'react-bootstrap';
import style from './Menu.css';

let emptyImage = "https://s3-media2.fl.yelpcdn.com/assets/2/www/img/dca54f97fb84/default_avatars/menu_medium_square.png";

const MenuItem = (props) => {
  return (
    <div>
      <hr/>
      <Media>
        <Media.Left align="middle">
          <img className={style.menuimage} width={70} height={70} src={emptyImage} alt="thumbnail"/>
        </Media.Left>
        <Media.Body>
          <Media.Heading componentClass="h5" bsClass="pull-left">
            <strong className={style.bluetext}>
              {props.details.item_name}
            </strong>
        </Media.Heading>
          <Media.Heading componentClass="h5" bsClass="pull-right">
            <strong>
              {props.details.item_price}
            </strong>
        </Media.Heading>
          <br></br>
          <br></br>
          <p>{props.details.item_description}</p>
        </Media.Body>
      </Media>
    </div>
  )
}
export default MenuItem
