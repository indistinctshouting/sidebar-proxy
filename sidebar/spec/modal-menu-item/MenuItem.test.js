import React from 'react';
import { shallow } from 'enzyme';
import MenuItem from '../../client/src/components/MenuItem';

describe('MenuItem', () => {
  let wrapper;
  
  beforeEach(() => {
    wrapper = shallow(
      <MenuItem
        details={{
          item_name: 'potato',
          item_price: '500.00',
          item_description: 'it is a potater',
        }}
      />
    )
  })

  it('should render a MenuItem correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a item name bolded in blue', () => {
    let itemName = wrapper.find('.blue-text');
    expect(itemName.text()).toMatch('potato');
  })

  it('should have the price bolded', () => {
    expect(wrapper.containsAnyMatchingElements([
      <strong>500.00</strong>
    ])).toEqual(true);
  })

  it('should have a description in a <p> tag', () => {
    let description = wrapper.find('p');
    expect(description.text()).toMatch('it is a potater');
  });
  
});