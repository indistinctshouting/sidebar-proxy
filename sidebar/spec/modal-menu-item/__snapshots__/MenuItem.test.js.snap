// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem should render a MenuItem correctly 1`] = `
<div>
  <hr />
  <Media
    bsClass="media"
    componentClass="div"
  >
    <MediaLeft
      align="middle"
      bsClass="media-left"
    >
      <img
        alt="thumbnail"
        height={70}
        src="https://s3-media2.fl.yelpcdn.com/assets/2/www/img/dca54f97fb84/default_avatars/menu_medium_square.png"
        width={70}
      />
    </MediaLeft>
    <MediaBody
      bsClass="media-body"
      componentClass="div"
    >
      <MediaHeading
        bsClass="pull-left"
        componentClass="h5"
      >
        <strong
          className="blue-text"
        >
          potato
        </strong>
      </MediaHeading>
      <MediaHeading
        bsClass="pull-right"
        componentClass="h5"
      >
        <strong>
          500.00
        </strong>
      </MediaHeading>
      <br />
      <br />
      <p>
        it is a potater
      </p>
    </MediaBody>
  </Media>
</div>
`;
